### PR TITLE
[stable10] Unified symfony event for updates happening in share

### PR DESCRIPTION
**Pull request by sharidas**
A unified symfony event for updates happening in share.
This change covers the updates for updates in:
a) share name
b) password change
c) update in expiry date
d) update of permission

Signed-off-by: Sujith H <sharidasan@owncloud.com>

## Description

A unified event to handle updates in share, like:
a) share name update
b) share password update
c) update of expiry date
d) permissions

## Motivation and Context

A unified event to handle updates in share, like:
a) share name update
b) share password update
c) update of expiry date
d) permissions

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: lib/private/Share20/Manager.php [hunk 1]
```diff
@@ -783,8 +787,11 @@ public function updateShare(\OCP\Share\IShare $share) {
 				'permissions' => $share->getPermissions(),
 				'path' => $userFolder->getRelativePath($share->getNode()->getPath()),
 			]);
+			$shareAfterUpdateEvent->setArgument('permissionupdate', true);
+			$shareAfterUpdateEvent->setArgument('path', $userFolder->getRelativePath($share->getNode()->getPath()));
 		}
 
+		$this->eventDispatcher->dispatch('share.afterupdate', $shareAfterUpdateEvent);
```
Comment (PVince81) [inline lib/private/Share20/Manager.php]: what if nothing got updated ?

File: lib/private/Share20/Manager.php [hunk 2]
```diff
@@ -749,13 +749,16 @@ public function updateShare(\OCP\Share\IShare $share) {
...
+		$shareAfterUpdateEvent = new GenericEvent(null);
+		$shareAfterUpdateEvent->setArgument('shareobject', $share);
 		if ($expirationDateUpdated === true) {
 			\OC_Hook::emit('OCP\Share', 'post_set_expiration_date', [
 				'itemType' => $share->getNode() instanceof \OCP\Files\File ? 'file' : 'folder',
 				'itemSource' => $share->getNode()->getId(),
 				'date' => $share->getExpirationDate(),
 				'uidOwner' => $share->getSharedBy(),
 			]);
+			$shareAfterUpdateEvent->setArgument('expiredateupdated', true);
```
Comment (PVince81) [inline lib/private/Share20/Manager.php]: the pattern for name was "oldname" and "newname", why not use "oldexpiredate" or "newexpiredate" then for consistency ? only "oldpassword" would not be possible

please settle on a single consistent format

File: lib/private/Share20/Manager.php [hunk 3]
```diff
@@ -783,8 +791,14 @@ public function updateShare(\OCP\Share\IShare $share) {
 				'permissions' => $share->getPermissions(),
 				'path' => $userFolder->getRelativePath($share->getNode()->getPath()),
 			]);
+			$shareAfterUpdateEvent->setArgument('permissionupdate', true);
```
Comment (PVince81) [inline lib/private/Share20/Manager.php]: "oldpermission" ?

**Comment by PVince81** [comment]
I wonder whether we really need the boolean flags. A listener can simply check whether there is a value "oldname" to know that the name was changed.

In an ideal object oriented world I'd want to have "oldShare" and "newShare" objects but I'm not sure you can get ahold of the old instance. Or would need a copy. I agree to stay with separate attributes for "oldXXX" then.

File: lib/private/Share20/Manager.php (hunk 3 above)
Comment (sharidas) [reply lib/private/Share20/Manager.php]: added 'oldpermissions'

**Comment by sharidas** [comment]
@PVince81 Ready for review.

**Comment by sharidas** [comment]
> I wonder whether we really need the boolean flags. A listener can simply check whether there is a value "oldname" to know that the name was changed.

The reason why I added boolean flag, is because it would be better to trigger event only when required.

File: lib/private/Share20/Manager.php (hunk 2 above)
Comment (PVince81) [reply lib/private/Share20/Manager.php]: "expiredate" vs "expirydate" vs "expirationdate"

three different words for the same thing: let's use the "expiration" variant

Comment (sharidas) [reply lib/private/Share20/Manager.php]: Updated the PR with "exipration"

**Review by PVince81 — APPROVED** [review]
:+1: thanks

**Comment by PVince81** [comment]
@sharidas please forward port to master (PR was missing "backport-request" label)

**Comment by sharidas** [comment]
Forward port to master : https://github.com/owncloud/core/pull/31138

**Pull request merged**